src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java

```diff
@@ -959,18 +964,28 @@ public void setSneaking(boolean sneaking)
 		this.sneaking = sneaking;
 	}
 
+	public void simulateSneak(boolean sneak)
+	{
+		PlayerToggleSneakEvent event = new PlayerToggleSneakEvent(this, sneak);
+		Bukkit.getPluginManager().callEvent(event);
```

Comment:
The method should respect the Event outcome and set the `sneaking` state accordingly.